src/spotless/groovy/net/neoforged/gradleutils/spotless/SpotlessUtilsExtension.groovy

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.gradleutils.spotless
+
+import com.diffplug.gradle.spotless.JavaExtension
+import com.diffplug.gradle.spotless.SpotlessExtension
+import com.diffplug.gradle.spotless.SpotlessTask
+import groovy.transform.CompileStatic
+import org.gradle.api.Project
+
+import javax.inject.Inject
+import java.nio.file.Files
+
+@CompileStatic
+abstract class SpotlessUtilsExtension {
+    final File configPath
+
+    @Inject
+    SpotlessUtilsExtension(Project project) {
+        configPath = new File(project.rootDir, '.gradle/formatter-config.xml')
+
+        final extract = project.tasks.register('extraGUSpotlessConfiguration', ExtractSpotlessConfiguration) {
+            it.output.set(configPath)
+        }
+
+        project.tasks.withType(SpotlessTask).configureEach {
+            it.dependsOn(extract)
+        }
+    }
+
+    void configure(SpotlessExtension ext) {
+        ext.java(this.&configure)
+    }
+
+    void configure(JavaExtension ext) {
+        // Spotless requires the config file to exist during configuration
+        if (!configPath.exists()) {
+            Files.createDirectories(configPath.toPath().parent)
+
+            try (
+                    final input = SpotlessUtilsExtension.getResourceAsStream('/formatter-config.xml');
+                    final output = Files.newOutputStream(configPath.toPath())
+            ) {
+                input.transferTo(output)
+            }
+        }
+
+        ext.endWithNewline()
```

Comment:
Potentially "no NotNull" too.

Reply:
I don't really like the idea of having non-auto-fixable rules by default.

Reply:
Javax also has `@Nonnull` and jetbrains has `@NotNull` so that also depends on what libraries the project has.

Reply:
(and not all projects generate package infos with the global `@XNonnull` annotations that also aren't standard)

Reply:
The package info is not necessary. NotNull is really a terrible pattern regardless of whether you generate the package infos or not. But I mostly care about eliminating it from FML, not so much about the other projects.